components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/APIMappingUtil.java

```diff
@@ -336,9 +325,14 @@ public static APIInfoDTO fromAPIToInfoDTO(API api) {
         for (String environmentName : environmentsPublishedByAPI) {
             Environment environment = environments.get(environmentName);
             if (environment != null) {
-                APIEnvironmentURLsDTO environmentURLsDTO = new APIEnvironmentURLsDTO();
-                String[] gwEndpoints = environment.getApiGatewayEndpoint().split(",");
-
+                APIURLsDTO apiurLsDTO = new APIURLsDTO();
+                APIDefaultVersionURLsDTO apiDefaultVersionURLsDTO = new APIDefaultVersionURLsDTO();
+                String[] gwEndpoints = null;
+                if(api.getType().equalsIgnoreCase("WS")) {
```

Comment:
Formatting issue. And shall we check the equals other way around to avoid NPE?

Reply:
Fixed